### PR TITLE
Updated misc Corp Type Codes

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* None

*Description of changes:*
- deleted obsolete NrRequestTypeCodes values
- updated some comments
- renamed CorpTypeCd.BC_CORPORATION -> CorpTypeCd.CORPORATION
- renamed CorpTypeCd.NR_SOLE_PROP -> CorpTypeCd.SOLE_PROPRIETORSHIP
- added CorpTypeCd.XPRO_LIMITED_LIABILITY_COMPANY
- added CorpTypeCd.SOCIETY_NR
- renamed CorpTypeCd.BC_UNLIMITED -> CorpTypeCd.UNLIMITED_LIABILITY_COMPANY
- added CorpTypeCd.XPRO_LIMITED_LIABILITY_PARTNERSHIP
- added CorpTypeCd.XPRO_LIMITED_PARTNERSHIP
- added CorpTypeCd.XPRO_SOCIETY_NR
- added CorpTypeCd.XPRO_UNLIMITED_LIABILITY_COMPANY

Ref: [request_types.pdf](https://github.com/bcgov/bcrs-shared-components/files/11985538/request_types.pdf)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/enums/name-request-codes.ts [hunk 1]
```diff
@@ -5,14 +5,6 @@
 
 // ref: https://github.com/bcgov/namex/blob/main/api/namex/constants/__init__.py
 export enum NrRequestTypeCodes {
-  // DEPRECATION WARNING - these will be removed!
-  BC_COMPANY = 'BC', // use NEW_BC instead
-  BC_CORPORATION = 'CR', // use NEW_CORP instead
-  BC_UNLIMITED = 'UL', // use NEW_ULC instead
-  COOP = 'CP', // use NEW_COOP instead
-  NR_SOLE_PROP = 'FR', // use NEW_FIRM instead
-  XPRO_CORPORATION = 'XCR', // use NEW_XPRO_CORP instead
```
Comment (severinbeauvais) [inline src/enums/name-request-codes.ts:14]: I couldn't find these used in Filings UI/Create UI/Edit UI/Auth Web/Namerequest UI (and there are alternate enums below with these values) so I deleted these.

File: src/enums/name-request-codes.ts [hunk 2]
```diff
@@ -35,13 +27,13 @@ export enum NrRequestTypeCodes {
...
-  NEW_FIRM = 'FR', // SP, DBA or GP
-  NEW_ULC = 'UL',
-  NEW_XPRO_CORP = 'XCR',
+  NEW_BC = 'BC', // BC Benefit Company Incorporation
+  NEW_CCC = 'CC', // CCC - Incorporation/Amalgamation
+  NEW_COOP = 'CP', // Cooperative - Incorporation/Amalgamation
+  NEW_CORP = 'CR', // BC Company - Incorporation/Amalgamation
+  NEW_FIRM = 'FR', // Sole Proprietorship/General Partnership/DBA - Registration
+  NEW_ULC = 'UL', // Unlimited Liability Company
+  NEW_XPRO_CORP = 'XCR', // Corporation (Foreign) - Extrapro Registration/Amalgamation
```
Comment (severinbeauvais) [inline src/enums/name-request-codes.ts:36]: I took these "descriptions" from the PDF linked in this PR.

File: src/modules/corp-type-module/corp-type-module.ts [hunk 3]
```diff
@@ -16,14 +16,14 @@ export enum CorpTypeCd {
   CCC_CONTINUE_IN = 'CCC',
   CEMETARY = 'CEM',
   COOP = 'CP',
-  BC_CORPORATION = 'CR', // SPECIAL NAMEREQUEST-ONLY ENTITY TYPE
+  CORPORATION = 'CR', // SPECIAL NAMEREQUEST-ONLY ENTITY TYPE
```
Comment (severinbeauvais) [inline src/modules/corp-type-module/corp-type-module.ts:19]: Renamed to better match the various descriptions (and not confuse it with `BC`).

File: src/modules/corp-type-module/corp-type-module.ts [hunk 4]
```diff
@@ -16,14 +16,14 @@ export enum CorpTypeCd {
...
   CONT_IN_SOCIETY = 'CS',
   ULC_CONTINUE_IN = 'CUL',
   DOING_BUSINESS_AS = 'DBA', // SPECIAL NAMEREQUEST-ONLY ENTITY TYPE
   EXTRA_PRO_REG = 'EPR',
   FINANCIAL = 'FI',
   FOREIGN = 'FOR',
-  NR_SOLE_PROP = 'FR', // SPECIAL NAMEREQUEST-ONLY ENTITY TYPE
+  SOLE_PROPRIETORSHIP = 'FR', // SPECIAL NAMEREQUEST-ONLY ENTITY TYPE
```
Comment (severinbeauvais) [inline src/modules/corp-type-module/corp-type-module.ts:26]: Renamed to better match the various descriptions.

File: src/modules/corp-type-module/corp-type-module.ts [hunk 5]
```diff
@@ -40,24 +40,30 @@ export enum CorpTypeCd {
...
+  XPRO_LIMITED_LIABILITY_COMPANY = 'RLC', // SPECIAL NAMEREQUEST-ONLY ENTITY TYPE
   RAILWAYS = 'RLY',
   SOCIETY = 'S',
   SOCIETY_BRANCH = 'SB',
+  SOCIETY_NR = 'SO', // SPECIAL NAMEREQUEST-ONLY ENTITY TYPE
   SOLE_PROP = 'SP',
   TRUST = 'T',
   TRAMWAYS = 'TMY',
-  BC_UNLIMITED = 'UL', // SPECIAL NAMEREQUEST-ONLY ENTITY TYPE
+  UNLIMITED_LIABILITY_COMPANY = 'UL', // SPECIAL NAMEREQUEST-ONLY ENTITY TYPE
```
Comment (severinbeauvais) [inline src/modules/corp-type-module/corp-type-module.ts:51]: Renamed to better match the various descriptions

**Comment by severinbeauvais** [comment]
As far as I can tell, the 3 enums that I renamed aren't used in any of our UIs (except the feature branch for Namerequest UI, where I will update it shortly).

This change will allow me to get rid of the "Other" enum in Namerequest UI and map EntityType entirely to CorpTypeCd. Ref: https://github.com/bcgov/namerequest/pull/622#issuecomment-1624201058

**Comment by severinbeauvais** [comment]
The EditShareStructure failing unit tests above pre-existed this PR. There is ticket [16965](https://app.zenhub.com/workspaces/entities-team-space-6143567664fb320019b81f39/issues/gh/bcgov/entity/16965) to fix them later.

**Review by cameron-eyds — APPROVED** [review]
Looks great!

**Pull request merged**